Missing stack trace with async mocha tests

Reproduced with NativeScript 2.0.
## Execute the following commands:

``` shell
$ tns create TestApp
$ tns test init
$ tns test ios --emulator
```
## Add the following test:

``` javascript
describe('Test', function () {
    it('Test', function (done) {
        setTimeout(function() {
            assert.equal(-1, 1);
            done();
        }, 1000);
    });
});
```
## The output is:

```
1   0xe0050 NativeScript::FFICallback<NativeScript::ObjCMethodCallback>::ffiClosureCallback(ffi_cif*, void*, void**, void*)
2   0x7a7c50 ffi_closure_inner
3   0x7a85a2 ffi_closure_i386
4   0x380c136 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__
5   0x380bb32 __CFRunLoopDoTimer
6   0x37ca39a __CFRunLoopRun
7   0x37c9846 CFRunLoopRunSpecific
8   0x37c965b CFRunLoopRunInMode
9   0x5573664 GSEventRunModal
10  0x55734a1 GSEventRun
11  0x1568eb9 UIApplicationMain
12  0x7a8468 ffi_call_i386
undefined: JS ERROR AssertionError: expected -1 to equal 1
```
## Expected

The JavaScript stack trace should be printed along with the `sourceURL` for the `AssertionError`.

Ping @ginev

--- a/src/NativeScript/JSErrors.mm
+++ b/src/NativeScript/JSErrors.mm
@@ -10,17 +10,18 @@
 
 #include <iomanip>
 #include <iostream>
+#include <sstream>
 
+#import "JSWarnings.h"
+#import "JSWorkerGlobalObject.h"
+#import "TNSRuntime+Diagnostics.h"
+#import "TNSRuntime+Inspector.h"
+#import "TNSRuntime+Private.h"
+#include "inspector/GlobalObjectConsoleClient.h"
+#include "inspector/GlobalObjectInspectorController.h"
 #include <JavaScriptCore/APICast.h>
 #include <JavaScriptCore/ScriptCallStack.h>
 #include <JavaScriptCore/ScriptCallStackFactory.h>
-#include "inspector/GlobalObjectInspectorController.h"
-#include "inspector/GlobalObjectConsoleClient.h"
-#import "TNSRuntime+Private.h"
-#import "TNSRuntime+Inspector.h"
-#import "TNSRuntime+Diagnostics.h"
-#import "JSWarnings.h"
-#import "JSWorkerGlobalObject.h"
 
 static TNSUncaughtErrorHandler uncaughtErrorHandler;
 void TNSSetUncaughtErrorHandler(TNSUncaughtErrorHandler handler) {
@@ -58,23 +59,25 @@ void reportFatalErrorBeforeShutdown(ExecState* execState, Exception* exception, 
             CFRunLoopRunInMode((CFStringRef)TNSInspectorRunLoopMode, 0.1, false);
         }
     } else {
-        std::cerr << "***** " << closingMessage << " *****\n";
+        NSLog(@"***** %s *****\n", closingMessage.operator const char*());
 
-        std::cerr << "Native stack trace:\n";
+        NSLog(@"Native stack trace:");
         WTFReportBacktrace();
 
-        std::cerr << "\nJavaScript stack trace:\n";
+        NSLog(@"JavaScript stack trace:");
         RefPtr<Inspector::ScriptCallStack> callStack = Inspector::createScriptCallStackFromException(execState, exception, Inspector::ScriptCallStack::maxCallStackSizeToCapture);
         for (size_t i = 0; i < callStack->size(); ++i) {
+            std::stringstream callstackLine;
             Inspector::ScriptCallFrame frame = callStack->at(i);
-            std::cerr << std::setw(4) << std::left << std::setfill(' ') << (i + 1) << frame.functionName().utf8().data() << "@" << frame.sourceURL().utf8().data();
+            callstackLine << std::setw(4) << std::left << std::setfill(' ') << (i + 1) << frame.functionName().utf8().data() << "@" << frame.sourceURL().utf8().data();
             if (frame.lineNumber() && frame.columnNumber()) {
-                std::cerr << ":" << frame.lineNumber() << ":" << frame.columnNumber();
+                callstackLine << ":" << frame.lineNumber() << ":" << frame.columnNumber();
             }
-            std::cerr << "\n";
+            NSLog(@"%s", callstackLine.str().c_str());
         }
 
-        std::cerr << "JavaScript error:\n";
+        NSLog(@"JavaScript error:");
+
         // System logs are disabled in release app builds, but we want the error to be printed in crash logs
         GlobalObjectConsoleClient::setLogToSystemConsole(true);
         globalObject->inspectorController().reportAPIException(execState, exception);

